pkg/ast/common.go

```diff
@@ -104,7 +104,7 @@ func ProcessSingleFilter(colName string, colValue interface{}, originalColValue
 						log.Errorf("qid=%d, ProcessSingleFilter: Failed to compile regex for %s. This may cause search failures. Err: %v", qid, t, err)
 						return nil, fmt.Errorf("invalid regex: %s", t)
 					}
-					criteria := CreateTermFilterCriteria(colName, compiledRegex, opr, qid, caseConversion)
+					criteria := CreateTermFilterCriteria(colName, compiledRegex, opr, qid, caseConversion, valueIsRegex)
```

Comment:
`valueIsRegex` is in `caseConversion` struct. You do not need to pass it as a param here.

Same for other places.

Reply:
addressed in the latest commit.